client/src/preview/store/file.slice.ts

```diff
@@ -12,28 +14,132 @@ const filesSlice = createSlice({
   name: 'files',
   initialState,
   reducers: {
-    addOrUpdateFile: (state, action: PayloadAction<FileState>) => {
+    addNewFile: (
```

Comment:
Please remove duplication and restructure these three reducers.

Reply:
Done. Now there is only the `addOrUpdateFile` reducer, capable to handle these three cases.